extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java

```diff
@@ -359,4 +360,21 @@ private boolean hasEntities(JpaModelBuildItem jpaModel) {
         return !jpaModel.getEntityClassNames().isEmpty();
     }
 
+    /**
+     * This registers a part of JTA Context Propagation to be unremovable, because our quarkus-hibernate-orm imports excludes
+     * quarkus-narayana, which would mark it as unremovable. This means that if by accident or other, the
+     * smallrye-context-propagation-jta module gets imported by the user, there will be an annoying warning thrown by ArC
+     * telling the user something went wrong because this class was removed. Marking it unremovable does not add a dependency to
+     * this class, but if it's present, it avoids it being removed, which is what we want, since if it's present, it will be
+     * used by SR-CP anyway, and it handles the case where JTA is not present (as is the case for Hibernate Reactive) so it's
```

Comment:
But why is SR-CP going to bother with it, when no single component is using it?

Reply:
Because there's no mechanism in MP-CP services to not load a plugin if it's in the classpath. If it's there, it's used. It's then up to the plugin to handle the case that it has no context to propagate, because JTA is not present.